### DISCUSSION TITLE
XDrip vibrate issue

### DISCUSSION THREAD

**Discussion by 3DRZ** [NightscoutFoundation/xDrip#3786] [Bug reports]
Hello,
My xdrip keeps vibrating from time to time. It happens always on the log entry: noise - changed from startup to low

How can i get rid of this?
![Screenshot_2024-11-27-19-09-04-45_2dd2c1bcb8f659cd696bcba2072f751b](https://github.com/user-attachments/assets/7ddf65dc-7995-4e79-a8f8-32e7991bf731)

**Comment by Navid200** [comment]
Was there no notification?
The next time it happens, can you look to see what notification you see?

**Reply by 3DRZ** [reply]
No notification, just this log entry

**Reply by Navid200** [reply]
I don't believe xDrip makes any sound or vibration other than through a notification.

What is the hardware data source set to?

**Reply by 3DRZ** [reply]
Source is set to "patched libre app". It receives the data throug juglucco.

But this behaviour was present also when I connected my Libre2 directly to xDrip.

Uninstalling xDrip resolves the problem. No more random vibrations.

So the vibration definitly comes from xDrip.

**Reply by Navid200** [reply]
Can you disable the following to see if it stops the vibration?

Settings > Alarms and Alerts > Other Alerts > Bad (noisy) Value Alerts

**Reply by 3DRZ** [reply]
I had this already turned off.
I have basically disabled all of my alerts as they are way to loud.

**Reply by Navid200** [reply]
Is the phone a Samsung?

**Reply by 3DRZ** [reply]
No, it's a OnePlus 8T with Android 14.0. Beside the vibration issue and the alert problems (snooze not working sometimes, high volume on alerts) xDrip is working great.

**Reply by Navid200** [reply]
Please go to `Settings` > `NFC Scan Features` 

Is `Use NFC feature` enabled?
Does it have to be considering you don't use it?  Or do you?

**Reply by 3DRZ** [reply]
This feature is disabled

**Reply by Navid200** [reply]
OK,

When I look at the xDrip code, the only instances I find where xDrip uses the vibrator not through a notification is in the NFC section.
There is no other case I can find.

@PhilTU Have you ever seen a case where xDrip causes vibration without bringing up a notification?

**Reply by PhilTU** [reply]
I have not noticed any but my phone is usually on vibrate.

**Reply by 3DRZ** [reply]
Hmm, so a Android 14 thing or something with my phone?

**Reply by PhilTU** [reply]
I don't know. I don't recall seeing lots of "Changed from Startup to Low" messages like you show above. Is that persistent ?

**Comment by 3DRZ** [comment]
![Screenshot_2024-11-30-18-47-49-48_2dd2c1bcb8f659cd696bcba2072f751b](https://github.com/user-attachments/assets/2333fea3-bd2b-4c36-809e-cf1c81b3abf6)
Yes the log is full of it. Thats why it's so annoying, as I keep looking on my phone because it seems i got a message or something - but mostly its xDrip going crazy.

**Reply by PhilTU** [reply]
@Navid200 is this expected behaviour ?

**Reply by Navid200** [reply]
That log means that the noise has gone down.  But, why it keeps repeating suggests that the data source is really messed up.
It means that xDrip does not successfully accept the new noise and does not record it as the new value.  So, it keeps detecting that noise has gone down.
This is just a speculation on my part.  I have never seen this behavior.

Regardless, there is no vibration triggered by xDrip when that log is generated.

**Reply by 3DRZ** [reply]
Is there a way to get rid of it? I basically don't need the xDrip algorithm or something. Basically I would need it to show only the values of the white line (values direct from the sensor)

I use xDrip only to show the values because it has a cleaner UI then juglucco.
![Screenshot_2024-11-30-19-24-50-65_2dd2c1bcb8f659cd696bcba2072f751b](https://github.com/user-attachments/assets/02ac54e8-5c0b-41b2-8c72-3633e69265b1)

**Reply by 3DRZ** [reply]
In fact there should be the possibility to disable this algorithm, as it shows values way to late. See screenshot here: I'm going fast downwards where xDrip draws a graph wichs seem going flat. Like 10 minutes later it will for shure go down. But why showing a value way higher when it has the data that I'm basically falling down...

Thats why I don't use xDrip as main source as this is dangerouse behaviour...
![Screenshot_2024-11-30-19-30-12-97_2dd2c1bcb8f659cd696bcba2072f751b](https://github.com/user-attachments/assets/c080dbae-707c-4304-9534-6b78a16e5bc6)

**Reply by PhilTU** [reply]
that looks like the smoothing, you can change the time period but it's partly an artefact of putting too many things in series in my opinion.

**Reply by PhilTU** [reply]
https://xdrip.readthedocs.io/en/latest/install/libre2patch/

**Reply by 3DRZ** [reply]
Thanks, I will give it a try.
In the "NFC Scan menu" under "Libre3" 😡
Who decided that this makes sense to put this setting in there...

**Reply by PhilTU** [reply]
None of this daisy chaining of apps makes sense to me. The change was a request and I guess someone did that for convenience or something.

Did you also see the 1 minute values option, that works with the actual patched app but I'm nit sure if Juggluco sends every minute

**Reply by 3DRZ** [reply]
I saw the option. But I have to enter in development mode to enable it. Juglucco sends the values every minute, yes.

But the change to 10 minutes already made it way better.
Thanks for that.

Yes, I think just using xDrip would resolve a lot of quirks, only problem is that my doctors need the readings in libreview (juglucco has a function to transmit them automatically).

So basically it removes the need to scan my sensor manually.
If xDrip implements the possibility to upload to libreview one day, I will get rid of any additional apps.

**Reply by 3DRZ** [reply]
Additional thing on the vibration issue:
I disabled "vibration" on the android notification settings for xDrip.
This fixes the issue for now.
![Screenshot_2024-11-30-21-18-11-42_8142442b4340f41fb3bd44535ad8f76b](https://github.com/user-attachments/assets/845ac072-3d30-44c3-a8ba-9e3351894d18)

**Reply by PhilTU** [reply]
sounds like a plan
